testlog/testlog.go

```diff
@@ -31,30 +30,24 @@ func Capture(t *testing.T) func() {
 
 type testLogWriter struct {
 	*testing.T
-	mu      sync.Mutex
+	mu      sync.RWMutex
 	stopped bool
-	ch      chan []byte
 }
 
 func (w *testLogWriter) Write(p []byte) (n int, err error) {
-	w.mu.Lock()
-	defer w.mu.Unlock()
-	if !w.stopped {
-		w.ch <- p
-		return len(p), nil
+	w.mu.RLock()
```

Comment:
Concurrent writes seem to be okay, so it's okay to use a read lock here just to synchronize with the stop function.